src/XMakeBuildEngine/BackEnd/BuildManager/BuildParameters.cs

```diff
@@ -320,6 +325,7 @@ private BuildParameters(BuildParameters other)
             _disableInProcNode = other._disableInProcNode;
             _logTaskInputs = other._logTaskInputs;
             _logInitialPropertiesAndItems = other._logInitialPropertiesAndItems;
+            _warningsAsErrors = other._warningsAsErrors == null ? null : new HashSet<string>(other._warningsAsErrors, StringComparer.OrdinalIgnoreCase);
```

Comment:
I just did some benchmarks.  If you have 10 warnings to set as errors and your build logs 10,000 warnings, the fastest is to use `HashSet<T>(StringComparer.OrdinalIgnoreCase)`.  I tried doing a case sensitive search but things placed in the set are upper case and the item being checked for is uppercased before the check but that was slower.  The `SortedSet<T>` was just a little bit slower.

In my scenario, it took 1.04 milliseconds to check 10,000 times for the 10 items.  I think the perf here is good enough but at least know we know. :emoji-for-the-'the-more-you-know'-public-service-annoncement: